cilium/CFP-36975-configuration-profiles.md

```diff
@@ -0,0 +1,121 @@
+    *   `--identity-allocation-mode=crd`
+    *   `--disable-endpoint-crd=true`
+    *   `--enable-cilium-endpoint-slice=false`
+
+    Note: This is only an example profile. It's not ready for use.
+
+## Creating a Profile
+
+*   Provide clear documentation for each profile, including:
+    *   Purpose and use case
+    *   Benefits and limitations (trade-offs)
+    *   Specific configuration options (flags) enabled/disabled
+    *   Installation instructions
+    *   Supported Cilium versions
+    *   Known issues
+*   Implement profile selection via Helm charts.
+
+## Implementation
+
+The implementation will leverage Cilium's Helm chart capabilities:
+
+1.  **Helm Chart Values:**  Each profile will be defined as a set of Helm values overrides.  These overrides will configure the necessary Cilium feature flags.  For example, the `values.yaml` for the "High-Scale" profile might include:
+
+    ```yaml
+    # profiles/high-scale.yaml
+    disableEndpointCRD: false
+    policyEnforcementMode: "never"
+    # ... other relevant settings ...
+    ```
+
+2.  **Profile Selection:** Users will select a profile during installation by referencing the appropriate values file:
+
+    ```bash
+    helm install cilium cilium/cilium --version 1.17.x \
+      --namespace kube-system \
+      --values profiles/high-scale.yaml
+      # Or:
+      # --set profile=high-scale  (if we add a 'profile' value to the main chart)
+    ```
+    The main `values.yaml` might include a top-level `profile` key, allowing users to select a profile with `--set profile=high-scale`.  This would then conditionally include the settings from the corresponding profile file.
+
+3.  **Documentation:** Each profile will have a dedicated page in the Cilium documentation, detailing its characteristics, installation instructions, incompatible features and functionality, and relevant configuration options.  This documentation will be part of the Cilium documentation repository.
+
+4. **Testing:**
+    *   **Unit Tests:** The Helm chart rendering will be unit-tested to ensure that the correct flags are set for each profile.
+    *   **Integration/E2E Tests:**  A new test suite will be created to validate the functionality and stability of each profile. These tests will run against a Kubernetes cluster with Cilium installed using the profile's Helm values. A subset of existing tests will run together with new per-profile tests that cover key features and use cases relevant to the profile.
+    *   **Continuous Integration:** Profile tests will be integrated into the Cilium CI pipeline to ensure that changes to Cilium do not break existing profiles.
+
+## Alternatives Considered
+
+*   **Individual Flags:**  As discussed in the "Motivation" section, this approach is complex and error-prone.
+*   **Separate Helm Charts:**  Creating separate Helm charts for each profile would lead to significant code duplication and maintenance overhead.
+
+## Open Issues
+
```

Comment:
should also consider adding guidelines on how users can upgrade from `basic-networking` profile to network policy enabled or other profiles without much disruption?

Reply:
I agree. It should be discussed and covered in the issue for `basic-networking` / `high-scale` profile - https://github.com/cilium/cilium/issues/37510